scripts/git-auth.sh

```
@@ -167,3 +164,14 @@ git_auth_env_for_url() {
   GIT_AUTH_LABEL="$token_label"
   GIT_AUTH_PROVIDER="$provider"
 }
+
+# Run a command with GIT_AUTH_ENV exported by the shell itself. The subshell
+# keeps the injected values scoped to this invocation, while avoiding an
+# external `env NAME=secret ...` process whose argument list exposes them.
+git_auth_run() (
+  local entry
+  for entry in ${GIT_AUTH_ENV[@]+"${GIT_AUTH_ENV[@]}"}; do
+    export "$entry"
```

Shellcheck (0.11.0)

[warning] This does not export 'entry'. Remove $/${} for that, or use ${var?} to quiet.

(SC2163)